platforms/nuttx/src/px4/common/usr_hrt.cpp

```diff
@@ -121,6 +107,43 @@ event_thread(int argc, char *argv[])
 	return 0;
 }
 
+static void start_worker(void)
+{
+	uintptr_t started;
+
+	// Don't use g_usr_hrt_task as the "lock" indicator, as this can result in
+	// two threads trying to create the worker
+	sched_lock();
```

Comment:
does sched_lock work when called from userspace?

Reply:
replying to self, it does but creates a syscall. Intentional? Would it be possible to check first if g_worker_started is already 1, then there is no need to lock the scheduler

Reply:
Yes, it bores a hole into kernel.

Reply:
> replying to self, it does but creates a syscall. Intentional? Would it be possible to check first if g_worker_started is already 1, then there is no need to lock the scheduler on every call

Yes, this is a good optimization and should be fixed.